### DISCUSSION TITLE
Why does the params definition change and why did it stop supporting undefined?

### DISCUSSION THREAD

**Discussion by KonradLinkowski** [mswjs/msw#1526] [Q&A]
In `0.39.2` the definition of `get` looked like that:
```
get: <RequestBodyType_2 extends DefaultRequestBody = DefaultRequestBody, Params_2 extends PathParams = PathParams, ResponseBody_2 extends DefaultRequestBody = DefaultRequestBody>(path: Path, resolver: ResponseResolver<RestRequest<never, Params_2>, RestContext, ResponseBody_2>) => RestHandler<import("./handlers/RequestHandler").MockedRequest<DefaultRequestBody>>;
```
params:
```
Params_2 extends PathParams = PathParams
```
`PathParams`:
```
export declare type PathParams = Record<string, string | ReadonlyArray<string>>;
```

In `1.0.0` the definition of get looks like that:
```
get: <RequestBodyType_2 extends DefaultBodyType = DefaultBodyType, Params_2 extends PathParams<keyof Params_2> = PathParams<string>, ResponseBody_2 extends DefaultBodyType = DefaultBodyType>(path: Path, resolver: ResponseResolver<RestRequest<never, Params_2>, RestContext, ResponseBody_2>) => RestHandler<MockedRequest<DefaultBodyType>>;
```
params:
```
Params_2 extends PathParams<keyof Params_2> = PathParams<string>
```
`PathParams`
```
type PathParams<KeyType extends keyof any = string> = {
    [ParamName in KeyType]: string | ReadonlyArray<string>;
 ```
 
 Now params cannot be marked as potentially `undefined`, because it throws a typescript error:
 ```
 type UpdateParams = {
    sort: string | undefined;
}
Type 'UpdateParams' does not satisfy the constraint 'PathParams<"sort">'.
  Types of property 'sort' are incompatible.
    Type 'string | undefined' is not assignable to type 'string | readonly string[]'.
      Type 'undefined' is not assignable to type 'string | readonly string[]'.
```
But it worked fine before.

Why do query params have to be defined now?

**Comment by kettanaito** [comment]
Hey, @KonradLinkowski. Thanks for reporting this. 

Could you please confirm this is still an issue for you on the latest version of MSW? `npm i msw@latest`. We've introduced quite a number of bug fixes since you reported this issue, there's a chance this has been fixed. Thanks.

**Reply by sgoridotla1** [reply]
@kettanaito, I am still facing the same issue.

```json
{
  "msw": "^2.0.11",
  "typescript": "^4.9.3"
}
```

**Reply by kettanaito** [reply]
Thanks! Looks like a bug to me. Path parameters should extend `undefined` since there's no guarantee they will be defined on runtime:

https://github.com/mswjs/msw/blob/ea28bd9ad2efe0918660cd37054f21883caa3d30/src/core/utils/matching/matchRequestUrl.ts#L6-L8

@sgoridotla1, would you be interested in opening a pull request to fix this?
